sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Catalog.scala

```diff
@@ -44,7 +90,20 @@ trait Catalog {
 
   def tableExists(tableIdentifier: Seq[String]): Boolean
 
-  def lookupRelation(
+  final def lookupRelation(
+      tableIdentifier: Seq[String],
+      alias: Option[String] = None): LogicalPlan = {
+    val tableIdent = processTableIdentifier(tableIdentifier)
+    if (tableIdent.length == 1 && tableIdent.head == TABLES_META_NAME) {
+      LocalRelation(tablesMetaOutput, tablesMeta().map(tablesMetaConverter))
+    } else if (tableIdent.length == 1 && tableIdent.head == COLUMNS_META_NAME) {
+      LocalRelation(columnsMetaOutput, columnsMeta().map(columnsMetaConverter))
+    } else {
+      lookupRelationInternal(tableIdentifier, alias)
+    }
+  }
+
+  protected def lookupRelationInternal(
```

Comment:
need some refactor here to git rid of this method, I'll try stackable trait later.